### ISSUE TITLE
Poor Accuracy When Evaluating the Model on My Own Dataset

### ISSUE THREAD

**Issue by amit154154** [AaronComo/LipFD#21]
I'm attempting to evaluate the model on a custom dataset containing real and fake videos (MP4 with audio). I've followed the preprocessing instructions provided in the repository as closely as possible, but the model consistently classifies nearly all real videos as fake.

 My Preprocessing Pipeline:
I applied the following preprocessing steps to my dataset:

Converted videos to exactly 25 FPS:

`ffmpeg -y -i input.mp4 -r 25 -c:v libx264 -preset fast -crf 23 -c:a aac output.mp4
`
Extracted 16 kHz mono WAV audio from videos:

`ffmpeg -y -i video.mp4 -vn -ac 1 -ar 16000 audio.wav
`

Cropped videos to 224×224 face-centered using MediaPipe with scale=1.2:

```
import cv2, mediapipe as mp
face_detection = mp.solutions.face_detection.FaceDetection(model_selection=0, min_detection_confidence=0.5)

def crop_video(input_video, output_video, scale=1.2, size=224):
    # OpenCV and MediaPipe logic to crop face region (centered square crop)
    pass  # simplified here for brevity
```

When evaluating, I consistently get poor results:

Accuracy ~31%

False Positive Rate (real videos predicted as fake) ~99%

Essentially, every real video is classified incorrectly as fake.

What I’ve Verified:

1.Videos have FPS exactly 25.

2.Audio is exactly 16 kHz mono.

3. Cropped video is 224×224 with face-centered crops (visually similar to your dataset).

I'm assuming there's a subtle preprocessing mismatch or an additional detail I might've missed. Could you confirm if there's a step or detail in preprocessing that I might be overlooking, or maybe could you give the code snippet for preprocessing video that is not in the repository?

 the image attached is of a real video from my dataset after the preprocessing .
Thanks for your support and the amazing research!
![Image](https://github.com/user-attachments/assets/fe093b61-4594-4a86-a1cb-d00901f328be)

**Comment by Mohand995** [comment]
F

**Comment by amit154154** [comment]
Hi, I think i found something interesting, I thought that the problem was that it predict every video as fake becouse i used the validate script and that is why i thought that there is a problem with the preprocssing.
but after more inspection it looks like there is a mistake in the validate code, specifactly this line
`tp, fn, fp, tn = cm.ravel()  `

after i looked at the docmentation it looks like the correct unpacking is 
`tn, fp, fn, tp = cm.ravel()`

which means that it predict almost every video as **REAL** which means that maybe I tested too advance of a lipsync mehods that the model didnlt generalize in and it is just has bad accuracy on it(when i tested the pre-processing i didn't cared much about what the deepfake videos were because i thought there were a problem with real videos which couldn't be any domain gap or something like that)

I am now testing on wav2lip and videos that are more in part with the training data and i will update in my resutls!

Also i am working on a script that inference on a video so i will also keep you posted on it.

**Comment by Mohand995** [comment]
can you share fake videos from wav2lip

**Comment by amit154154** [comment]
i just did a test on a few videos (about 80 preprocessed frames, like in the first messege) on wav2lip and got 66% accaurcy, it is not great but it is an improvement that suggest that the preprocessing really wasn't the issue.
i will later do a larger test that would be more conclusive.

**Comment by FastWind123** [comment]
I have tested on my own dataset and it also tends to predict every video as **REAL**.

**Comment by FastWind123** [comment]
And after I switch to easier datasets, where fake video use less effective lipsyc algorithm, the accuracy enhances, just the same as you.

**Comment by mribrahim** [comment]
I am not sure how much the model is using the audio inconsistency for prediction, it is probably using the facial features to distinguish fake or real.

**Comment by guyrose3** [comment]
@amit154154 were you able to reach high accuracy on real data using this model? either using it as is or by retraining/finetuning with higher quality fakes
